### DISCUSSION TITLE
Optional Image Origin Checks for Static Next.js Sites? Balancing Security and Convenience

### DISCUSSION THREAD

**Discussion by Josekariz** [community/community#131103] [Programming Help] [answered]
### Select Topic Area

Question

### Body

**Current Behavior:**

> Next.js prioritizes security with strict image origin checks. While this is great for most apps, it can be a hassle for those of us using it for static sites with limited user interaction.

**The Challenge:**

> As a relatively new user, I find it challenging to use external images due to these security measures. This impacts my workflow and makes it harder to build static sites efficiently.

**Proposed Solution:**

> Configurable Flag: Introducing a flag in next.config.js to disable strict image origin checks for static sites. This flag would come with a clear warning about potential security implications.

What do you all think? Would an optional flag to make using external images easier for static sites be helpful? I'd love to hear your thoughts, experiences, and any alternative solutions you might have.

**Comment by paiineli** [comment]
Hi,

I completely understand your concerns regarding Next.js's strict image origin checks, especially when building static sites. Your proposed solution of introducing a configurable flag in next.config.js to disable these checks could indeed streamline the workflow for developers working with external images.

Benefits of Your Proposal:
Flexibility: It would provide users the flexibility to choose their security level based on the project needs.
Efficiency: For static sites, this could significantly reduce the friction in using external images, making development more efficient.

Considerations:
Security Implications: As you mentioned, a clear warning about the potential risks of disabling these checks is crucial. Developers should be aware of the trade-offs involved.
Community Feedback: Gathering feedback from other users could help gauge the demand for such a feature and refine its implementation.

Overall, I believe this flag could be a valuable addition for those of us building static sites. I'd love to hear what others think or if anyone has alternative solutions that have worked for them!

Best,

**Comment by ghost** [comment]
Hi, you don't actually have to use this, but if you remove it, then you will have a lot of vulnerabilities, of course, even this can be fixed, but you need to do everything carefully.

**Reply by Josekariz** [reply, marked as answer]
LOL, Thanks, so I'll do it as the Romans in Next.
